### ISSUE TITLE
Send PUT requests instead of PATCH requests

### ISSUE THREAD

**Issue by RubenVerborgh** [solid/query-ldflex#16]
Unfortunately, `PATCH` requests are broken on node-solid server 4.x and 5.x in at least two ways:
1. node-solid-server does not follow the SPARQL UPDATE spec when a DELETE clause when there are no or multiple matches (https://github.com/solid/node-solid-server/issues/1085), so the client gets unexpected 409 responses. 
2. node-solid-server does not implement file locking (https://github.com/solid/node-solid-server/issues/137), so accidentally concurrent patches can destroy a file, so the client gets unexpected 500 responses.

The first thing is more or less easily fixable, but deciding about the right fix is a longer process (https://github.com/linkeddata/rdflib.js/pull/299). But even if we fix it, the client doesn't know whether it is talking to a fixed or a legacy server, and both require different approaches.

The second thing is hard to fix, as code for writing files is spread all over node-solid-server.

Hence, HTTP `PATCH` is currently not a reliable way forward. As such, I propose to:
- apply the patch locally on the client, and update the remote resource through `PUT`
- use the eTag mechanism to ensure we are not overwriting other people's changes

This solution is not ideal because it is inefficient for bandwidth, and will cause performance problems with larger files. However, it avoids file corruption, which is what we are currently risking, and it does not require changes in the client code (which was not broken in the first place).

TL;DR: `PATCH` is broken on the server, but we cannot easily fix it there; instead of forcing a workaround in clients, we can put that workaround in LDflex.

**Comment by rubensworks** [comment]
We could implement this in LDflex, but disable it by default behind a config option, which could then be enabled automatically when using query-ldflex.

**Comment by RubenVerborgh** [comment]
Postponed because of https://github.com/solid/node-solid-server/issues/1085#issuecomment-464892054.
